sstables/compaction.cc

```diff
@@ -440,7 +440,7 @@ class compaction {
     utils::UUID _run_identifier;
     ::io_priority_class _io_priority;
     // optional clone of sstable set to be used for expiration purposes, so it will be set if expiration is enabled.
-    std::optional<sstable_set> _sstable_set;
+    lw_shared_ptr<sstable_set> _sstable_set;
```

Comment:
We want the sstable_set to be wrapped by a shared pointer because all the snapshots of it also need to own it. We aren't actually creating snapshots of the _sstable_set in compaction, so in this case we could leave the lw_shared_ptr but I'm still not sure if that's the best solution

Reply:
Can't it be done internally in sstable_set?

Reply:
It is now